core/src/main/java/feign/RequestTemplate.java

```diff
@@ -535,7 +536,16 @@ public RequestTemplate headers(Map<String, Collection<String>> headers) {
    * @see Request#headers()
    */
   public Map<String, Collection<String>> headers() {
-    return Collections.unmodifiableMap(headers);
+
+    return Collections.unmodifiableMap(
+        headers.entrySet().stream().filter(h -> h.getValue() != null && !h.getValue().isEmpty())
+            .collect(toMap(
```

Comment:
just use default toMap... `toMap(Entry::getKey, Entry::getValue)` no need to handle duplicated keys or map creation

Reply:
if I do so, it returns the map in any order, with LinkedHashMap it collects them in same order. In fact that was my first approach and it broke tests